pkg/kv/kvserver/allocator_scorer.go

```diff
@@ -827,10 +956,15 @@ func candidateListForRemoval(
 		}
 		diversityScore := diversityRemovalScore(s.StoreID, existingStoreLocalities)
 		candidates = append(candidates, candidate{
-			store:          s,
-			valid:          constraintsOK,
-			necessary:      necessary,
-			fullDisk:       !maxCapacityCheck(s),
+			store:     s,
+			valid:     constraintsOK,
+			necessary: necessary,
+			fullDisk:  !maxCapacityCheck(s),
+			// When removing a replica from a store, we do not want to include
+			// high amplification in ranking stores. This would submit already
+			// high read amplification stores to additional load of moving a
+			// replica.
+			highReadAmp:    false,
```

Comment:
@shralex @lidorcarmel This is where logic would be placed to shed replicas when considering read amp.